homeassistant/components/media_extractor/__init__.py

```diff
@@ -106,7 +108,19 @@ def extract_and_send(self) -> None:
 
     def get_stream_selector(self) -> Callable[[str], str]:
         """Return format selector for the media URL."""
-        ydl = YoutubeDL({"quiet": True, "logger": _LOGGER})
+        cookies_file = os.path.join(
+            self.hass.config.config_dir, "media_extractor_cookies.txt"
+        )
+        if os.path.isfile(cookies_file):
+            ydl = YoutubeDL(
+                {"quiet": True, "logger": _LOGGER, "cookiefile": cookies_file}
```

Comment:
Can we maybe just update the options dict we pass in if its a file instead of duplicating the creation of the object